src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -342,7 +447,7 @@ private async Task<Expression> RecurseChainedExpression(ChainedExpression expres
         {
             var feedOptions = new QueryRequestOptions
             {
-                MaxConcurrency = -1, // execute counts across all partitions
+                MaxConcurrency = MaxQueryConcurrency, // execute counts across all partitions
```

Comment:
Should this remain as -1, or should the comment be updated?

Reply:
Good catch, I'll update the comment